tracing/trace_state.hh

```diff
@@ -505,28 +505,113 @@ private:
     friend void stop_foreground_prepared(const trace_state_ptr& state, const cql3::query_options* prepared_options_ptr) noexcept;
 };
 
-class trace_state_ptr final {
+
+class opentelemetry_state final {
 private:
     lw_shared_ptr<trace_state> _state_ptr;
+    bool const _opentelemetry_tracing{false};
```

Comment:
Per my previous comment, it should be dynamically filled in (not const!)

Reply:
I think making it const makes it similar to "classic" tracing. As you said "There it works this way: if the user requested "classic" tracing on a request, it builds non-empty trace_state_ptr. Otherwise, an "empty" trace_state_ptr is passed around.". The decision if opentelemetry is used is made somewhere at the beginning of request processing.

Reply:
Yes, what I meant is that the problem is that the `_openetelemetry_tracing` can never be `true` - hardcoded to `false` and not properly filled in.